duktape/src/main/java/com/squareup/duktape/DuktapeException.java

```diff
@@ -15,8 +15,58 @@
  */
 package com.squareup.duktape;
 
+import android.text.TextUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public final class DuktapeException extends RuntimeException {
+  /**
+   * Duktape stack trace strings have multiple lines of the format "function filename.ext:line".
+   * "function" is optional, but we'll omit those frames from the exception, since it means the
+   * frame is in native code.
+   */
+  private final static Pattern STACK_TRACE_PATTERN =
+      Pattern.compile("\\s*([^\\s]+) ([^\\s]+):(\\d+).*");
```

Comment:
should this have a trailing `$` so the line is at the end? I'm wondering about hypothetical filenames like `hello:123.js`